contracts/multisig-prover/src/encoding/evm.rs

```diff
@@ -55,6 +55,18 @@ impl TryFrom<Signer> for Operator {
     }
 }
 
+impl BatchID {
+    pub fn new(message_ids: &[String]) -> BatchID {
+        let mut message_ids = message_ids
+            .iter()
+            .map(|id| id.as_bytes())
+            .collect::<Vec<&[u8]>>();
+        message_ids.sort();
+
+        Keccak256::digest(message_ids.concat()).as_slice().into()
+    }
```

Comment:
point of discussion: message ids are in a sense user input. Is it possible to create two batches with the same id but different messages?

Reply:
if the content of the batches is different, the workers would have rejected one set of messages or the other during voting